onnx/test/function_inference_test.py

```diff
@@ -73,6 +76,31 @@ def test_fi_attribute(self):
         dtype_10 = onnx.helper.make_attribute("dtype", 10)
         self._check(code, [float_type_], [dtype_10], [float16_type_])
 
+    def test_fi_optional_input1(self):
```

Comment:
maybe a more descriptive test name (than `input1`)?

Reply:
The description is `optional_input`. I have two test-cases, so added suffixes 1 and 2. The comments describe the test-cases.

Reply:
I recommend we split each group (behavior) into its own test case. For example, the first test can be named as “test_fi_omits_type_for_missing_trailing_optional_parameter” and so on. 

This way each test is focused on a single behavior, and it is super clear in the logs when any test fails.

Reply:
Hmmm. I personally disagree. Trying to capture the full intention of a test in the function-name can't be met 100%, and I think that's ok. I feel the suggested name goes overboard. I prefer using comments for a more complete description.

Reply:
I think it can be hard to capture 100% of the intention in a test name, and that’s ok. It is good to still do the exercise. When we can’t, it is usually a good hint that we may be able to test a smaller unit (of the logic or the behavior). 

This way of naming (`test_<method or function name>_<expected behavior>_[when_<condition>]`) encourages that we create tests that will validate a particular behavior (and in many cases that behavior only). So that we have cleanly separated, easy to understand test cases. 

In this particular case, I can see parameterizing the input creation part of the logic useful so that we don’t need to duplicate the model texts.

Reply:
Fair point. Let me clarify what I mean further. First, IMHO, this is a low-priority. Eg: I would love to see the items in [this issue](https://github.com/onnx/onnx/issues/3651) addressed ... they are higher priority because they impact a much wider audience (every backend that implements onnx and every frontend/exporter that generates onnx) when some aspect of an onnx operator spec is unclear. Yet, that remains mostly unaddressed, and has for a long time. Second, it is more important to ensure complete coverage of all behavior we want to test. Breaking them into units as you suggest may sometimes incur an overhead (it is even possible to end up with misleading names because precisely capturing the expected behavior and the conditions in a single identifier is often tricky). So, I will leave it as is. (I actually ended up merging the two test-cases into one, at least now we don't have artificial names with "1" and "2".)